src/formatters/diffs/string.js

```diff
@@ -0,0 +1,143 @@
+import Consts from './const';
+
+const DIFF_MARKER_SYMBOL  = '^';
+const EMPTY_STRING_LENGTH = '""'.length;
+
+var cutNewLines = function (code) {
+    return typeof code === 'string' ? code.replace(/(\r\n|\n|\r)/gm, '\\n') : code;
+};
+
+var getMarkerStr = function (offset) {
+    return [(new Array(offset + 1)).join(' '),
```

Comment:
We need recalculate offset for marker after adding the array index. Otherwise we'll need to do it in templates.